mitodl/ol-infrastructure
# Upgrade Pydantic to V2 and migrate existing models

# What are the relevant tickets?
Closes #1610 

# Description (What does it do?)
Pydantic released a major version upgrade from v1 to v2 which includes a number of breaking changes. (Reference post
[here](https://docs.pydantic.dev/latest/blog/pydantic-v2/) and migration guide [here](https://docs.pydantic.dev/latest/migration/)). A portion of the upgrades were managed through the use of [`bump-pydantic`](https://github.com/pydantic/bump-pydantic), followed by manually trying to execute existing usage of these models throughout ol-infrastructure.

# How can this be tested?
This has been manually tested by executing various Concourse pipelines that make use of Pydantic models, as well as locally building a Concourse Vagrant instance through Pydantic. Once this merges we will see more widespread application of the code paths through automatic runs of Concourse pipelines.
